req_compile/cmdline.py

```diff
@@ -548,38 +551,48 @@ def _generate_repo_header(
 
 
 def _write_index_directives(repos: Sequence[Repository], write_to: IO[str]) -> None:
-    """Write the --index-url and --extra-index-url lines in the requirement files.
+    """Write the `--index-url`, `--extra-index-url`, and `--find-links` lines in the requirement files.
 
     Args:
         repos: All repos used in the solution.
         write_to: Output to write to.
     """
-    wrote_any = False
+    index_content = ""
+    links_content = ""
     for repo in repos:
         if isinstance(repo, PyPIRepository) and repo.index_type != IndexType.DEFAULT:
-            wrote_any = True
-            write_to.write(str(repo) + "\n")
+            index_content += str(repo) + "\n"
+        elif isinstance(repo, FindLinksRepository):
```

Comment:
Most people would use --find-links as a way to do offline compiles but don't consider that directory portable along with the compiled requirements.  I think it would be surprising to see this in the output file, especially since it can be an abs path on someones computer.

There is a flag for omitting directives, maybe that should be an enum.  NONE, INDEXES, ALL

Reply:
Updated!